pkg/cmd/workload_run.go

```diff
@@ -5,87 +5,135 @@ import (
 	"errors"
 
 	"github.com/cloud-native-application/rudrx/pkg/oam"
+	"github.com/cloud-native-application/rudrx/pkg/plugins"
 
 	"github.com/cloud-native-application/rudrx/api/types"
 	"github.com/cloud-native-application/rudrx/pkg/cmd/util"
-	"github.com/cloud-native-application/rudrx/pkg/plugins"
 
 	"github.com/spf13/cobra"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-const Staging = "staging"
-const App = "app"
+const (
+	Staging      = "staging"
+	App          = "app"
+	WorkloadType = "type"
+	TraitDetach  = "detach"
+)
 
 type runOptions oam.RunOptions
 
 func newRunOptions(ioStreams util.IOStreams) *runOptions {
 	return &runOptions{IOStreams: ioStreams}
 }
 
-func AddWorkloadCommands(parentCmd *cobra.Command, c types.Args, ioStreams util.IOStreams) error {
-	templates, err := plugins.LoadInstalledCapabilityWithType(types.TypeWorkload)
-	if err != nil {
-		return err
+func AddCompCommands(c types.Args, ioStreams util.IOStreams) *cobra.Command {
+	compCommands := &cobra.Command{
+		Use:                   "comp <commands>",
+		DisableFlagsInUseLine: true,
+		Short:                 "Manage Components",
+		Long:                  "Manage Components",
+		Annotations: map[string]string{
+			types.TagCommandType: types.TypeWorkloads,
+		},
```

Comment:
Now it needs to take two steps to create an application?

Reply:
vela comp run -t deployment myapp